wheatley/row_generation/helpers.py

```diff
@@ -29,7 +29,10 @@ def convert_pn(pn_str: str) -> List[List[int]]:
 
     symmetric = pn_str.startswith('&')
 
-    cleaned = re.sub("[.]*[x-][.]*", ".-.", pn_str).strip('.& ').split('.')
+    # The 'replace' fixes issue #88, whereby the string 'xx' will be converted to 'x..x', thus resulting
+    # in incorrect expansion of methods with consecutive 'x's.  It is, however, a giant hack and we should
+    # probably re-write this parser and fix it properly.
+    cleaned = re.sub("[.]*[x-][.]*", ".-.", pn_str).strip('.& ').replace('..', '.').split('.')
```

Comment:
I think an alternative to using `.replace('..', '.').split('.')` would be to split this into an intermediate variable (which wouldn't be a bad thing to do anyway) and use
```
re.split([.]+, replaced_string)
```
But don't mind which one seems easier to read to you

Reply:
Updated!